plugin/evm/atomic_trie.go

```diff
@@ -182,10 +181,13 @@ func lastCommittedRootIfExists(db database.Database) (common.Hash, uint64, error
 		return common.Hash{}, 0, nil
 	case err != nil:
 		return common.Hash{}, 0, err
-	case len(lastCommittedHeightBytes) != wrappers.LongLen:
-		return common.Hash{}, 0, fmt.Errorf("expected value of lastCommittedKey to be %d but was %d", wrappers.LongLen, len(lastCommittedHeightBytes))
 	}
-	height := binary.BigEndian.Uint64(lastCommittedHeightBytes)
+
+	height, err := database.ParseUInt64(lastCommittedHeightBytes)
+	if err != nil {
+		return common.Hash{}, 0, fmt.Errorf("expected value length of lastCommittedKey to be %d but was %d: %w", wrappers.LongLen, len(lastCommittedHeightBytes), err)
+	}
```

Comment:
```suggestion
		return common.Hash{}, 0, fmt.Errorf("expected value at lastCommittedKey to be a valid uint64: %w", err)
	}
```

it seems odd to add a formatted error message that infers exactly what went wrong inside of `database.ParseUint64` instead of including that in the error message returned by `database.ParseUint64`.